io.openems.edge.controller.ess.cycle/src/io/openems/edge/controller/ess/cycle/ControllerEssCycleImpl.java

```diff
@@ -84,31 +92,62 @@ protected void deactivate() {
 
 	@Override
 	public void run() throws OpenemsNamedException {
-
-		// get max charge/discharge power
-		var maxDischargePower = this.ess.getPower().getMaxPower(this.ess, Phase.ALL, Pwr.ACTIVE);
-		var maxChargePower = this.ess.getPower().getMinPower(this.ess, Phase.ALL, Pwr.ACTIVE);
-
-		// store current state in StateMachine channel
-		var state = this.stateMachine.getCurrentState();
-		this.channel(ControllerEssCycle.ChannelId.STATE_MACHINE).setNextValue(state);
-
-		// Prepare Context
-		var previousState = this.stateMachine.getPreviousState();
-		var context = new Context(this, this.config, this.componentManager, this.ess, maxChargePower, maxDischargePower,
-				previousState, this.lastStateChange);
-
-		// Call the StateMachine
-		try {
-			this.stateMachine.run(context);
-		} catch (OpenemsNamedException e) {
-			this.logError(this.log, "StateMachine failed: " + e.getMessage());
+		if (this.parsedStartTime == null) {
+			this.logError(this.log, "Start time could not be parsed");
+			return;
 		}
-
-		if (previousState != state) {
-			// State has changed
-			this.lastStateChange = LocalDateTime.now(this.componentManager.getClock());
+		switch (this.config.mode()) {
+		case MANUAL_ON:
+			// get max charge/discharge power
+			var allowedDischargePower = this.ess.getPower().getMaxPower(this.ess, Phase.ALL, Pwr.ACTIVE);
+			var allowedChargePower = this.ess.getPower().getMinPower(this.ess, Phase.ALL, Pwr.ACTIVE);
+
+			// Prepare Context
+			var context = new Context(this, //
+					this.config, //
+					this.componentManager.getClock(), //
+					this.ess, //
+					allowedChargePower, //
+					allowedDischargePower, //
+					this.parsedStartTime);
+
+			// store current state in StateMachine channel
+			var currentState = this.stateMachine.getCurrentState();
+			this._setStateMachine(currentState);
```

Comment:
Usually we do this as first command in the `handleStateMachine` logic. (see e.g. https://github.com/OpenEMS/openems/blob/develop/io.openems.edge.ess.generic/src/io/openems/edge/ess/generic/symmetric/EssGenericManagedSymmetricImpl.java#L100)

Downside of your approach is, that the State will never get touched if mode is switched to `MANUAL_OFF`. Imagine a UI Widget that displays the State of the Controller: it would still show "START_DISCHARGE", even though the Controller actually does nothing.